azavea/tilegarden
# Feature/extend aws options

## Overview

This PR extends the configurability of AWS Lambda on deployment by adding more options to `.env`. New options include:
 * `LAMBDA_REGION`: AWS region to which you are deploying
 * `LAMBDA_ROLE`: IAM role given to your lambdas
 * `LAMBDA_TIMEOUT`: Time in seconds before your lambdas time out
 * `LAMBDA_MEMORY`: Memory in MB allocated to your lambdas
 * `LAMBDA_SUBNETS`: VPC Subnets used to configure VPC
 * `LAMBDA_SECURITY_GROUPS`: VPC Security Groups used to configure VPC

The first two options are required on all new deployments, the rest are optional. 


### Notes
Only `LAMBDA_TIMEOUT` and `LAMBDA_MEMORY` are changed on "update" deployments, all other settings can only be changed by deploying a new lambda (or by logging in to the Lambda Management Console).


## Testing Instructions

 * Create a new `.env` file based on the contents of `env-template`.
 * Set up and deploy a new lambda, or update an old one.
 * Change the value of one of the settings variables and re-deploy.
 * Log in to the Lambda Management Console and you should see that the value of that setting has updated.
 

Resolves #34
